### PR TITLE
adding initial commit for the details page

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/containers/RepositoriesDetailContainer/RepositoriesDetailContainer.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import _ from 'lodash';
+
+// Actions
+import * as actionCreators from './RepositoriesDetailDuck.js';
+
+// Components
+import RepoDetail from '../../components/RepoDetail/';
+
+const mapStateToProps = (state, props) => {
+  console.log('state', state);
```
Comment (vitkon) [inline src/containers/RepositoriesDetailContainer/RepositoriesDetailContainer.js]: can we remove console.log here?

File: src/containers/RepositoriesDetailContainer/RepositoriesDetailContainer.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    id: props.id
+  };
+};
+
+const mapDispatchToProps = dispatch => bindActionCreators(actionCreators, dispatch);
+
+class RepositoriesDetailContainer extends Component {
+
+  static propTypes = {
+    repos: PropTypes.array,
```
Comment (vitkon) [inline src/containers/RepositoriesDetailContainer/RepositoriesDetailContainer.js]: array or undefined?

File: src/containers/RepositoriesDetailContainer/RepositoriesDetailContainer.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  };
+
+  constructor (props) {
+    super(props);
+    props.load();
+  }
+
+  render () {
+    let repoId = parseInt(this.props.id);
+    console.log('test', this.props);
```
Comment (vitkon) [inline src/containers/RepositoriesDetailContainer/RepositoriesDetailContainer.js]: console.log

File: src/containers/RepositoriesDetailContainer/RepositoriesDetailContainer.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    const repo = _.find(this.props.repos, {'id': repoId});
```
Comment (vitkon) [inline src/containers/RepositoriesDetailContainer/RepositoriesDetailContainer.js:35]: `.filter` could've been used here, too

File: src/containers/RepositoriesDetailContainer/RepositoriesDetailContainer.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    load: PropTypes.func.isRequired
+  };
+
+  constructor (props) {
+    super(props);
+    props.load();
+  }
+
+  render () {
+    let repoId = parseInt(this.props.id);
```
Comment (vitkon) [inline src/containers/RepositoriesDetailContainer/RepositoriesDetailContainer.js]: why `let`?

File: src/containers/RepositoriesDetailContainer/RepositoriesDetailDuck.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  [LOAD]: (state, action) => ({
+    ...state,
+    isLoading: true
+  }),
+  [LOAD_SUCCESS]: (state, action) => ({
+    ...state,
+    isLoading: false,
+    repos: action.data.map(repo => ({
+      author: repo.owner.login,
+      avatar: repo.owner.avatar_url,
```
Comment (vitkon) [inline src/containers/RepositoriesDetailContainer/RepositoriesDetailDuck.js:24]: how do we ensure that the data in repo that we receive is consistent?
e.g. are we 100% sure that `repo.owner` object is going to be there?

@michael-martin data from API needs to go through a sanitisation layer (middleware?), after that step we should not need null checks.

File: src/containers/RepositoriesDetailContainer/RepositoriesDetailDuck.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+export default function reducer (state = initialState, action = {}) {
+  const handler = REDUCERS[action.type];
+  return handler ? handler(state, action) : state;
+};
+
+// Action Creators
+export function load () {
+  return (dispatch, getState) => {
+    // Already loading?
```
Comment (vitkon) [inline src/containers/RepositoriesDetailContainer/RepositoriesDetailDuck.js:47]: is this comment needed? Self explanatory imfo

File: src/containers/RepositoriesDetailContainer/RepositoriesDetailDuck.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    if (getState().repositories.isLoading) {
+      return;
+    }
+
+    // Notify that we're loading.
```
Comment (vitkon) [inline src/containers/RepositoriesDetailContainer/RepositoriesDetailDuck.js:52]: same here, no need for comment, vars and methods name need to express the intention

File: src/routes/Detail/Detail.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+import RepoDetail from '../../containers/RepositoriesDetailContainer/';
+
+export const Detail = (props) => (
+  <div className="row">
+    <div className="col-xs-12">
+      <RepoDetail id={props.params.id} />
+    </div>
+  </div>
+);
+
```
Comment (vitkon) [inline src/routes/Detail/Detail.js:18]: propTypes please

File: src/containers/RepositoriesDetailContainer/RepositoriesDetailDuck.js (hunk 7 above)
Comment (ghost) [inline src/containers/RepositoriesDetailContainer/RepositoriesDetailDuck.js:47]: As a starter kit, especially for people getting used to react & redux. I think it's handy to have these comments?

Comment (michael-martin) [inline src/containers/RepositoriesDetailContainer/RepositoriesDetailDuck.js:47]: Agreed with @girwin575 here. One of the things I plan to do with this is add extensive comments here. This code is as much about teaching as anything else! :)

File: src/routes/Detail/index.js [hunk 10]
```diff
@@ -0,0 +1,2 @@
+import component from './Detail.js';
+export default component;
```
Comment (michael-martin) [inline src/routes/Detail/index.js]: There's a one-liner version of this now, can you swap to that?

`export { default } from './Detail.js'`

File: src/containers/RepositoriesDetailContainer/RepositoriesDetailDuck.js (hunk 7 above)
Comment (vitkon) [inline src/containers/RepositoriesDetailContainer/RepositoriesDetailDuck.js:47]: fine. you got away with this one ;)

File: src/components/RepoDetail/RepoDetail.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
+import React, { PropTypes } from 'react';
+
+const RepoDetail = (props) => {
+  console.log('xxx', props);
```
Comment (vitkon) [inline src/components/RepoDetail/RepoDetail.js:4]: can we remove console.log here?

**Pull request merged**